seatunnel-connectors-v2/connector-iotdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/iotdb/source/IoTDBSource.java

```diff
@@ -0,0 +1,140 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.seatunnel.connectors.seatunnel.iotdb.source;
+
+import static org.apache.seatunnel.connectors.seatunnel.iotdb.config.SourceConfig.FIELDS;
+import static org.apache.seatunnel.connectors.seatunnel.iotdb.config.SourceConfig.HOST;
+import static org.apache.seatunnel.connectors.seatunnel.iotdb.config.SourceConfig.NODE_URLS;
+import static org.apache.seatunnel.connectors.seatunnel.iotdb.config.SourceConfig.PORT;
+import static org.apache.seatunnel.connectors.seatunnel.iotdb.constant.SourceConstants.FIELDS_K_V_SPLIT;
+import static org.apache.seatunnel.connectors.seatunnel.iotdb.constant.SourceConstants.FIELDS_SPLIT;
+
+import org.apache.seatunnel.api.common.PrepareFailException;
+import org.apache.seatunnel.api.common.SeaTunnelContext;
+import org.apache.seatunnel.api.source.Boundedness;
+import org.apache.seatunnel.api.source.SeaTunnelSource;
+import org.apache.seatunnel.api.source.SourceReader;
+import org.apache.seatunnel.api.source.SourceSplitEnumerator;
+import org.apache.seatunnel.api.table.type.BasicType;
+import org.apache.seatunnel.api.table.type.SeaTunnelDataType;
+import org.apache.seatunnel.api.table.type.SeaTunnelRow;
+import org.apache.seatunnel.api.table.type.SeaTunnelRowType;
+import org.apache.seatunnel.common.config.CheckConfigUtil;
+import org.apache.seatunnel.common.config.CheckResult;
+import org.apache.seatunnel.common.constants.PluginType;
+import org.apache.seatunnel.connectors.seatunnel.iotdb.state.IoTDBSourceState;
+
+import org.apache.seatunnel.shade.com.typesafe.config.Config;
+
+import com.google.auto.service.AutoService;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@AutoService(SeaTunnelSource.class)
+public class IoTDBSource implements SeaTunnelSource<SeaTunnelRow, IoTDBSourceSplit, IoTDBSourceState> {
+
+    private SeaTunnelContext seaTunnelContext;
+
+    private SeaTunnelRowType typeInfo;
+
+    private Map<String, Object> configParams = new HashMap();
+
+    @Override
+    public String getPluginName() {
+        return "IoTDB";
+    }
+
+    @Override
+    public void prepare(Config pluginConfig) throws PrepareFailException {
+        CheckResult result = CheckConfigUtil.checkAllExists(pluginConfig, HOST, PORT);
+        if (!result.isSuccess()) {
+            result = CheckConfigUtil.checkAllExists(pluginConfig, NODE_URLS);
+
+            if (!result.isSuccess()) {
+                throw new PrepareFailException(getPluginName(), PluginType.SOURCE, "host and port and node urls are both empty");
+            }
+        }
+        if (pluginConfig.getString(FIELDS) == null) {
+            throw new PrepareFailException(getPluginName(), PluginType.SOURCE, "fields is empty");
+        }
+        this.typeInfo = convertToType(pluginConfig.getString(FIELDS));
```

Comment:
Can you support #2392? We should use the same standard to define the `schema` or `format`